tensorflow/cc/gradients/math_grad_test.cc

```diff
@@ -679,6 +682,30 @@ TEST_F(CWiseUnaryGradTest, Atan_Complex) {
   }
 }
 
+TEST_F(CWiseUnaryGradTest, Lgamma) {
+  auto x_fn = [this](const int i) { return RV({0, -1, 1, -2, 2, -3, 3}); };
+  auto dy_fn = [this](const float x) { return x + RV({-2, 2, -3, 3, -4, 4}); };
+  auto dx_fn = [this](const float x, const float dy) {
+    // where is digamma? Eigen::digamma ?
```

Comment:
Hi, @kbsriram and @skye . I get in trouble. Since `digamma` function is missing in std library, is [Eigen::digamma](https://eigen.tuxfamily.org/dox/namespaceEigen.html#af40db84b3db19fe25fe2f77c429420e5) a good candidate? Any help will be greatly appreciated, thanks.

Reply:
Hm... don't have any guidance here, though it strikes me all these tests probably should use ComputeGradientError anyway :-) but unfortunately it currently only tests for real-valued inputs, not complex ones.

In the interests of simplicity, maybe one alternative is to use `NaryGradTest` rather than the `CWiseUnaryGradTest`, and add a note that it doesn't test complex values? You'll want to select values away from any poles to avoid numeric estimation errors. Maybe something like
```c++
TEST_F(NaryGradTest, Lgamma) {
// ...
  TensorShape shape({2, 3});
  auto x = PlaceHolder(scope_, DT_FLOAT, Placeholder::Shape(shape));
  auto y = Lgamma(scope_, x);
  Tensor x_init_value = test::AsTensor<float>({ /* six random float values, but not close to any poles for Lgamma */}, {2, 3});
  RunTest(x, x_init_value, y, shape);
}

Reply:
Thank you very much! I'll modify code later.

Reply:
modified.